StackExchange.Profiling/AsyncWebRequestProfilerProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Threading;
+
+namespace StackExchange.Profiling
+{
+    public class AsyncWebRequestProfilerProvider : WebRequestProfilerProvider
+    {
+        protected override MiniProfiler Current
+        {
+            get
+            {
+                return _currentProfiler.Value;
+            }
+            set
+            {
+                _currentProfiler.Value = value;
+            }
+        }
+
+        protected override Timing CurrentHead
+        {
+            get
+            {
+                return _currentTiming.Value;
+            }
+            set
+            {
+                _currentTiming.Value = value;
+            }
+        }
+
+        private readonly AsyncLocal<MiniProfiler> _currentProfiler = new AsyncLocal<MiniProfiler>();
```

Comment:
btw, if we wanted to stay compatible with .net 4 we could use our RevAsyncLocal here instead. functionally it is pretty much the same and available as long as CallContext.SetLogicalData is available.  

If we are to suggest a pr for miniprofiler v4 we definitely will need to think about a substitution for AsyncLocal since they try to be backward compatible there